mav_trajectory_generation/include/mav_trajectory_generation/impl/polynomial_optimization_nonlinear_impl.h

```diff
@@ -192,45 +202,105 @@ int PolynomialOptimizationNonLinear<_N>::optimizeTimeMellingerOuterLoop() {
-                                           segment_times.size());
+    if (optimization_parameters_.print_debug_info_time_allocation) {
+      std::cout << "Segment times after opt: ";
+      for (const double seg_time : segment_times) {
+        std::cout << seg_time << " ";
+      }
+      std::cout << std::endl;
+      std::cout << "Final cost: " << final_cost << std::endl;
+      std::cout << "Nlopt result: " << result << std::endl;
+    }
+  }
 
   // Scaling of segment times
-  Eigen::VectorXd x = x_rel_change;
-  scaleSegmentTimesWithViolation(&x);
+  std::vector<double> relative_segment_times;
+  poly_opt_.getSegmentTimes(&relative_segment_times);
+  scaleSegmentTimesWithViolation();
+  std::vector<double> scaled_segment_times;
+  poly_opt_.getSegmentTimes(&scaled_segment_times);
 
   // Print all parameter after scaling
   if (optimization_parameters_.print_debug_info_time_allocation) {
-    std::cout << "[MEL          Original]: " << x_orig.transpose()
-              << std::endl;
-    std::cout << "[MEL RELATIVE Solution]: " << x_rel_change.transpose()
-              << std::endl;
-    std::cout << "[MEL          Solution]: " << x.transpose() << std::endl;
-    std::cout << "[MEL   Trajectory Time] Before: " << x_orig.sum()
-              << " | After Rel Change: " << x_rel_change.sum()
-              << " | After Scaling: " << x.sum()
+    std::cout << "[MEL          Original]: ";
+    std::for_each(original_segment_times.cbegin(),
+                  original_segment_times.cend(),
+                  [](double c) { std::cout << c << " "; });
+    std::cout << std::endl;
+    std::cout << "[MEL RELATIVE Solution]: ";
+    std::for_each(relative_segment_times.cbegin(),
+                  relative_segment_times.cend(),
+                  [](double c) { std::cout << c << " "; });
+    std::cout << std::endl;
+    std::cout << "[MEL          Solution]: ";
+    std::for_each(scaled_segment_times.cbegin(), scaled_segment_times.cend(),
+                  [](double c) { std::cout << c << " "; });
+    std::cout << std::endl;
+    std::cout << "[MEL   Trajectory Time] Before: "
+              << std::accumulate(original_segment_times.begin(),
+                                 original_segment_times.end(), 0)
+              << " | After Rel Change: "
+              << std::accumulate(relative_segment_times.begin(),
+                                 relative_segment_times.end(), 0)
+              << " | After Scaling: "
+              << std::accumulate(scaled_segment_times.begin(),
+                                 scaled_segment_times.end(), 0)
               << std::endl;
   }
 
   return result;
 }
 
 template <int _N>
-double PolynomialOptimizationNonLinear<_N>::getCostAndGradient(
-        std::vector<double>* gradients) {
+double PolynomialOptimizationNonLinear<_N>::getCost() const {
+  return poly_opt_.computeCost();
+}
 
-  // Weighting terms for different costs
-  const double w_d = 100.0;
```

Comment:
I assume this was just an arbitrary number? Or is this the "aggresivity" constant?

Reply:
Just a scaling factor, but didn't make any sense here since it was the only cost + only gradient being evaluated (not mixed with anything).

Reply:
Thanks for cleaning up then!